### PR TITLE
Add validation for configurable keys passed to .with_config()

### PR DESCRIPTION
**Pull request by nfcampos**
- Fix some typing issues found while doing that

### CODE REVIEW

File: libs/langchain/langchain/runnables/openai_functions.py [hunk 1]
```diff
@@ -4,7 +4,7 @@
 from typing_extensions import TypedDict
 
 from langchain.output_parsers.openai_functions import JsonOutputFunctionsParser
-from langchain.schema.output import ChatGeneration
```
Comment (nfcampos) [inline libs/langchain/langchain/runnables/openai_functions.py:7]: this was wrong

File: libs/langchain/langchain/schema/runnable/_locals.py [hunk 2]
```diff
@@ -11,7 +11,7 @@
     Union,
 )
 
-from langchain.schema.runnable.base import Input, Output, RunnableSerializable
+from langchain.schema.runnable.base import Input, Other, Output, RunnableSerializable
```
Comment (nfcampos) [inline libs/langchain/langchain/schema/runnable/_locals.py:14]: this is required after making Input contravariant

File: libs/langchain/langchain/schema/runnable/passthrough.py [hunk 3]
```diff
@@ -22,7 +22,7 @@
 
 from langchain.pydantic_v1 import BaseModel, create_model
 from langchain.schema.runnable.base import (
-    Input,
```
Comment (nfcampos) [inline libs/langchain/langchain/schema/runnable/passthrough.py:25]: same as above

File: libs/langchain/langchain/schema/runnable/base.py [hunk 4]
```diff
@@ -2296,6 +2296,25 @@ class RunnableBinding(RunnableSerializable[Input, Output]):
...
 
+    def __init__(
+        self,
+        *,
+        bound: Runnable[Input, Output],
+        kwargs: Mapping[str, Any],
+        config: Optional[Mapping[str, Any]] = None,
+        **other_kwargs: Any,
+    ) -> None:
+        config = config or {}
```
Comment (eyurtsev) [inline libs/langchain/langchain/schema/runnable/base.py:2307]: A doc-string could help here... this is a bit "non-local" -- i.e., would be difficult for another developer why this validation code makes sense here

**Review by eyurtsev — APPROVED** [review]

**Pull request merged**